docs/dev/guide/custom-aa-tutorial.md

```diff
@@ -0,0 +1,637 @@
+
+In this tutorial we build a factory that deploys 2-of-2 multisig accounts.
+
+## Preliminaries
+
+It is assumed that you are already familiar with deploying smart contracts on zkSync. If not, please refer to the first section of the [Hello World](./hello-world.md) tutorial and have read the [introduction](../zksync-v2/system-contracts.md) to the system contracts.
+
+It is also assumed that you already have some experience working with Ethereum.
+
+## Installing dependencies
+
+We will use the zkSync hardhat plugin for developing this contract. Firstly, we should install all the dependencies for it: 
+
+```
+mkdir cross-chain-tutorial
+cd cross-chain-tutorial
+yarn init -y
+yarn add -D typescript ts-node ethers zksync-web3 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy
+```
+
+Since we are working with zkSync contracts, we also need to install the package with the contracts and its peer dependencies:
+
+```
+yarn add @matterlabs/zksync-contracts @openzeppelin/contracts @openzeppelin/contracts-upgradeable
+```
+
+## Account abstraction
+
+Each account needs to implement the [IAccountAbstraction](https://github.com/matter-labs/v2-testnet-contracts/blob/07e05084cdbc907387c873c2a2bd3427fe4fe6ad/l2/system-contracts/interfaces/IAccountAbstraction.sol#L7) interface. Since we are building an account with signers, we should also have [EIP1271](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/83277ff916ac4f58fec072b8f28a252c1245c2f1/contracts/interfaces/IERC1271.sol#L12) implemented.
+
+So the skeleton for the contract will look the following way:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import '@matterlabs/zksync-contracts/l2/system-contracts/interfaces/IAccountAbstraction.sol';
+import '@matterlabs/zksync-contracts/l2/system-contracts/TransactionHelper.sol';
+
+import "@openzeppelin/contracts/interfaces/IERC1271.sol";
+
+contract TwoUserMultisig is IAccountAbstraction, IERC1271 {
+
+	modifier onlyBootloader() {
+		require(msg.sender == BOOTLOADER_FORMAL_ADDRESS, "Only bootloader can call this method");
+		// Continure execution if called from the bootloader.
+		_;
+	}
+
+    function validateTransaction(Transaction calldata _transaction) external payable override onlyBootloader {
+        _validateTransaction(_transaction);
+    }
+
+    function _validateTransaction(Transaction calldata _transaction) internal {
+        
+    }
+
+    function executeTransaction(Transaction calldata _transaction) external payable override onlyBootloader {
+		_executeTransaction(_transaction)
+	}
+
+    function _executeTransaction(Transaction calldata _transaction) internal {
+        
+    }
+
+    function executeTransactionFromOutside(Transaction calldata _transaction) external payable {
+		_validateTransaction(_transaction);
+		_executeTransaction(_transaction);
+	}
+
+
+	// bytes4(keccak256("isValidSignature(bytes32,bytes)")
```

Comment:
Did you mean to leave this here but commented out?

Reply:
Yes, it just shows how the constant was derived. Maube we should say "EIP1271 magic constant"

Reply:
I'll probably just remove it

Reply:
I think it's good